lib/src/container/mod.rs

```diff
@@ -333,10 +333,15 @@ impl<'a> ManifestDiff<'a> {
 /// as root, then a default isolation of running the process via `nobody` will be applied.
 pub fn merge_default_container_proxy_opts(
     config: &mut containers_image_proxy::ImageProxyConfig,
+    transport: Transport,
 ) -> Result<()> {
-    let user = rustix::process::getuid()
+    let mut user = rustix::process::getuid()
         .is_root()
```

Comment:
I think this would be a bit cleaner as
```
let user = (rustix::process::getuid().is_root && transport != Transport::ContainerStorage).then_some(isolation::DEFAULT_UNPRIVILEGED_USER);
``` 

or so?

Reply:
took the previous approach